hilti/runtime/CMakeLists.txt

```diff
@@ -74,19 +74,59 @@ foreach ( lib hilti-rt hilti-rt-debug )
     # jrx-objects which is an OBJECT library itself and depended on by
     # `jrx-objects`.
     target_link_libraries(${lib} jrx-objects)
-endforeach ()
 
-# Build hilti-rt with release flags.
-string(REPLACE " " ";" cxx_flags_release ${CMAKE_CXX_FLAGS_RELEASE})
-target_compile_options(hilti-rt-objects     PRIVATE ${cxx_flags_release})
-target_compile_options(hilti-rt-objects     PRIVATE "-g;-O3;-DNDEBUG;-Wall")
-target_compile_definitions(hilti-rt-objects PRIVATE "HILTI_RT_BUILD_TYPE_RELEASE")
-
-# Build hilti-rt-debug with debug flags.
-string(REPLACE " " ";" cxx_flags_debug ${CMAKE_CXX_FLAGS_DEBUG})
-target_compile_options(hilti-rt-debug-objects     PRIVATE ${cxx_flags_debug})
-target_compile_options(hilti-rt-debug-objects     PRIVATE "-UNDEBUG;-O0;-Wall")
-target_compile_definitions(hilti-rt-debug-objects PRIVATE "HILTI_RT_BUILD_TYPE_DEBUG")
+    if ("${lib}" MATCHES "debug$")
+        string(REPLACE " " ";" cxx_flags_debug ${CMAKE_CXX_FLAGS_DEBUG})
+        target_compile_options(hilti-rt-debug-objects     PRIVATE ${cxx_flags_debug})
+        target_compile_options(hilti-rt-debug-objects     PRIVATE "-UNDEBUG;-O0;-Wall")
+        target_compile_definitions(hilti-rt-debug-objects PRIVATE "HILTI_RT_BUILD_TYPE_DEBUG")
+    else ()
+        string(REPLACE " " ";" cxx_flags_release ${CMAKE_CXX_FLAGS_RELEASE})
+        target_compile_options(hilti-rt-objects     PRIVATE ${cxx_flags_release})
+        target_compile_options(hilti-rt-objects     PRIVATE "-g;-O3;-DNDEBUG;-Wall")
+        target_compile_definitions(hilti-rt-objects PRIVATE "HILTI_RT_BUILD_TYPE_RELEASE")
+    endif ()
```

Comment:
Why did this block have to move into the loop?

Reply:
Since we set up the precompiled headers in that loop we need all flags for the object libraries to be set up (the flags of for the precompilation are derived from them). If these flags are added after the loop, they will not be present in the flags of the precompiled headers.

The alternative would be to move the creation of the precompiled headers out of the loop, to some later point where all object library flags are set up.